airbyte-integrations/connectors/source-dixa/source_dixa/source.py

```diff
@@ -24,77 +24,66 @@
+class DixaStream(HttpStream, ABC):
 
-    @staticmethod
-    def datetime_to_ms_timestamp(dt: datetime) -> int:
-        """
-        Converts a datetime object to a millisecond-precision timestamp.
-        """
-        return int(dt.timestamp() * 1000)
+    primary_key = "id"
+    url_base = "https://exports.dixa.io/v1/"
 
-    @staticmethod
-    def add_days_to_ms_timestamp(days: int, milliseconds: int) -> int:
-        return ConversationExport.datetime_to_ms_timestamp(
-            ConversationExport.ms_timestamp_to_datetime(ConversationExport._validate_ms_timestamp(milliseconds)) + timedelta(days=days)
-        )
+    backoff_sleep = 60  # seconds
+
+    def __init__(self, config: Mapping[str, Any]) -> None:
+        super().__init__(authenticator=config["authenticator"])
+        self.start_date = datetime.strptime(config["start_date"], "%Y-%m-%d")
+        self.start_timestamp = utils.datetime_to_ms_timestamp(self.start_date)
+        self.end_timestamp = utils.datetime_to_ms_timestamp(datetime.now()) + 1
+        self.batch_size = config["batch_size"]
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
         return None
 
     def request_params(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
         self.logger.info(
-            f"Sending request with updated_after={stream_slice['updated_after']} and " f"updated_before={stream_slice['updated_before']}"
+            f"Sending request with updated_after={stream_slice['updated_after']} and updated_before={stream_slice['updated_before']}"
         )
         return stream_slice
 
-    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        yield from response.json()
-
     def backoff_time(self, response: requests.Response):
         """
-        The rate limit is 10 requests per minute, so we sleep for one minute
-        once we have reached 10 requests.
+        The rate limit is 10 requests per minute, so we sleep
+        for defined backoff_sleep time (default is 60 sec) before we continue.
 
         See https://support.dixa.help/en/articles/174-export-conversations-via-api
         """
-        return 60
+        return self.backoff_sleep
+
+
+class IncrementalDixaStream(DixaStream):
+
+    cursor_field = "updated_at"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield from response.json()
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, any]:
+        """
+        Uses the `updated_at` field, which is a Unix timestamp with millisecond precision.
+        """
+        current_stream_state = current_stream_state or {}
+        return {self.cursor_field: max(current_stream_state.get(self.cursor_field, 0), latest_record.get(self.cursor_field, 0))}
```

Comment:
maybe add start_date instead of 0 ?

Reply:
This is the `get_updated_state` we should have return the state value after we sync the stream.
We add `start_date` inside the `stream_slices` if we don't have the state we use `start_date`.